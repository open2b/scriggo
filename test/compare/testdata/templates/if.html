{# render #}

{# bool #}
{% if true %}if true{% end %}
{% if not true %}bad{% else %}(if not true){% end %}
{% if false %}bad{% else %}(if false){% end %}
{% if not false %}if not false{% end %}

{# integers #}
{% if 42 %}if 42{% end %}
{% if -300 %}if -300{% end %}
{% if 0 %}bad{% else %}(if 0){% end %}
{% if 0 + 0 %}bad{% else %}(if 0 + 0){% end %}

{# strings #}
{% if "something" %}if "something"{% end %}
{% if not "something" %}bad{% else %}(if not "something"){% end %}
{% s1 := "" %}{% if s1 %}bad{% else %}(if s1){% end %}
{% s2 := "" %}{% if not s2 %}if not s2{% end %}

{# slices #}
{% if []int{1, 2, 3} %}if []int{1, 2, 3}{% end %}
{% if []int{} %}if []int{}{% end %}
{% if []int(nil) %}bad{% else %}([]int(nil)){% end %}

{# interfaces #}
{% if interface{}(nil) %}bad{% else %}(interface{}(nil)){% end %}
{% if not interface{}(nil) %}not interface{}(nil){% end %}
{% if interface{}(5) %}interface{}(5){% end %}

{# pointers #}
{% if (*int)(nil) %}bad{% else %}((*int)(nil)){% end %}

{% if "" %}
	bad
{% else if "" %}
	bad
{% end %}

{% if 0 %}
	bad
{% else if 42 %}
    else if 42
{% end %}