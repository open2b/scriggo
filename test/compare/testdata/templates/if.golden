if true
(if not true)
(if false)
if not false

if 42
if -300
(if 0)
(if 0 + 0)

if "something"
(if not "something")
(if s1)
if not s2

if []int{1, 2, 3}
if []int{}
([]int(nil))


    else if 42